packages/shared/src/components/inputs/TokenAmountWithSliderInput.svelte

```diff
@@ -63,6 +71,47 @@
         }
         rawAmount = bigAmount.toString()
     }
+
+    function getInputLength(): number {
+        const length = rawAmount?.length || 1
+        const isDecimal = rawAmount?.includes('.') || rawAmount?.includes(',')
+
+        return length - (isDecimal ? 0.5 : 0)
+    }
+
+    function getFontSizeForInputLength(): string {
+        if (inputLength < 10) {
+            return '64'
+        } else if (inputLength < 14) {
+            return '48'
+        } else {
+            return '32'
+        }
+    }
+
+    function getMaxAmountOfDigits(): number {
+        const metadata = token?.metadata
+        if (!metadata) {
+            return 32
+        }
+
+        const decimalSeparator = getDecimalSeparator()
+
+        const decimalPlacesAmount = rawAmount?.includes(decimalSeparator)
+            ? rawAmount.split(decimalSeparator)[1].length || 1
+            : 0
+        const allowedDecimalAmount = Math.min(decimalPlacesAmount, metadata.decimals)
+
+        const integerLengthOfBalance =
+            formatTokenAmountDefault(availableBalance, metadata).split(decimalSeparator)?.[0]?.length ?? 0
+
+        return (
+            allowedDecimalAmount +
+            integerLengthOfBalance +
+            (metadata.decimals ? 1 : 0) +
+            (rawAmount?.includes(decimalSeparator) ? 1 : 0)
+        )
+    }
```

Comment:
These functions are duplicated in `TokenAmountInput`. Find a way to remove this duplication :relaxed:

Reply:
As discussed on the stand-up, this component needs to change for Governance anyway, so I'll approve it for now.